src/RadixDlt.NetworkGateway.PostgresIntegration/LedgerExtension/PostgresLedgerExtenderService.cs

```diff
@@ -281,281 +281,297 @@ private async Task<ExtendLedgerReport> ProcessTransactions(ReadWriteDbContext db
-                        CoreModel.EntityType.FungibleResource => typeof(FungibleResourceEntity),
-                        CoreModel.EntityType.NonFungibleResource => typeof(NonFungibleResourceEntity),
-                        _ => throw new ArgumentOutOfRangeException(),
-                    });
-                }
+                        var referencedEntity = referencedEntities
+                            .GetOrAdd(
+                                (EntityAddress)newGlobalEntity.EntityAddress,
+                                _ => new ReferencedEntity((EntityAddress)newGlobalEntity.EntityAddress,  newGlobalEntity.EntityType, stateVersion)
+                            );
 
-                foreach (var substate in stateUpdates.CreatedSubstates.Concat(stateUpdates.UpdatedSubstates).ToList())
-                {
-                    var substateId = substate.SubstateId;
-                    var substateData = substate.SubstateData;
+                        referencedEntity.WithTypeHint(newGlobalEntity.EntityType switch
+                        {
+                            CoreModel.EntityType.GlobalGenericComponent => typeof(GlobalGenericComponentEntity),
+                            CoreModel.EntityType.GlobalPackage => typeof(GlobalPackageEntity),
+                            CoreModel.EntityType.GlobalEpochManager => typeof(EpochManagerEntity),
+                            CoreModel.EntityType.GlobalClock => typeof(GlobalClockEntity),
+                            CoreModel.EntityType.GlobalValidator => typeof(GlobalValidatorEntity),
+                            CoreModel.EntityType.GlobalAccessController => typeof(GlobalAccessControllerEntity),
+                            CoreModel.EntityType.GlobalVirtualEd25519Account => typeof(GlobalAccountEntity),
+                            CoreModel.EntityType.GlobalVirtualSecp256k1Account => typeof(GlobalAccountEntity),
+                            CoreModel.EntityType.GlobalAccount => typeof(GlobalAccountEntity),
+                            CoreModel.EntityType.GlobalIdentity => typeof(GlobalIdentityEntity),
+                            CoreModel.EntityType.GlobalVirtualEd25519Identity => typeof(GlobalIdentityEntity),
+                            CoreModel.EntityType.GlobalVirtualSecp256k1Identity => typeof(GlobalIdentityEntity),
+                            CoreModel.EntityType.GlobalFungibleResource => typeof(GlobalFungibleResourceEntity),
+                            CoreModel.EntityType.GlobalNonFungibleResource => typeof(GlobalNonFungibleResourceEntity),
+                            _ => throw new ArgumentOutOfRangeException(nameof(newGlobalEntity.EntityType), newGlobalEntity.EntityType.ToString()),
+                        });
+                    }
 
-                    if (substateData is CoreModel.EpochManagerSubstate epochManagerSubstate)
+                    foreach (var substate in stateUpdates.CreatedSubstates.Concat(stateUpdates.UpdatedSubstates).ToList())
```

Comment:
What's the point of `.ToList()`?

Reply:
removed. I think i just forgot to remove it after debugging.